spec/datadog/tracing/contrib/support_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'tempfile'
+
+RSpec.describe Datadog::Tracing::Contrib::Support do
+  describe '.fully_loaded?' do
+    let(:temp_file) do
+      file = Tempfile.create(['autoloaded_constant', '.rb'])
+      file.write('module AutoloadedParent::AutoloadedConstant; end')
+      file.flush
+      file
+    end
+    let(:test_module) do
+      temp_file = self.temp_file
+      stub_const(
+        'AutoloadedParent',
+        Module.new do
+          autoload :AutoloadedConstant, temp_file.path
+          const_set(:LoadedConstant, Class.new)
+        end
+      )
+    end
+
+    after { temp_file.close }
```

Comment:
Should we go another way with `ensure` instead?

```ruby
around do |example|
  # tempfile creation
  example.run
ensure
  # tempfile close
end
```

or maybe we can go another way with FakeFS? https://github.com/fakefs/fakefs